### PR TITLE
feat(inoculate): `cargo`-style logs with `tracing`

### PR DESCRIPTION
**Pull request by hawkw**
This branch adds a hacked-together `tracing` formatter for
cargo-esque logging in `inoculate`.

Some screenshots:
image
image
image

### CODE REVIEW

File: inoculate/src/trace.rs [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+{
+    fn format_event(
+        &self,
+        ctx: &FmtContext<'_, S, N>,
+        mut writer: Writer<'_>,
+        event: &Event<'_>,
+    ) -> fmt::Result {
+        let metadata = event.metadata();
+        let level = metadata.level();
+        let color = ColorMode::default().should_color_stdout();
```
Comment (mystor) [inline inoculate/src/trace.rs:27]: Why the change to `should_color_stdout()` instead of `should_color_stderr()`? I suppose the old cargo formatter used `stderr` and tracing logs to `stdout`?

File: inoculate/src/trace.rs [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    }
+}
+
+impl Visit for Visitor<'_> {
+    fn record_debug(&mut self, field: &Field, value: &dyn fmt::Debug) {
+        if self.is_empty {
+            if self.level >= Level::INFO && field.name() == Self::MESSAGE {
+                let message = format!("{:?}", value);
+                if let Some((tag, message)) = message.as_str().split_once(' ') {
+                    if tag.len() <= Self::INDENT && tag.ends_with("ing") || tag.ends_with('d') {
```
Comment (mystor) [inline inoculate/src/trace.rs:102]: this code is definitely not sketchy at all :-)

I suppose doing it with separate fields for the action would be somewhat trickier as `MESSAGE` is usually the first field visited?

File: inoculate/src/trace.rs [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+                }
+                (Level::TRACE, true) => {
+                    write!(self.writer, "{}{} ", "trace".purple().bold(), ":".bold())
+                }
+                (Level::ERROR, false) => self.writer.write_str("error: "),
+                (Level::WARN, false) => self.writer.write_str("warning: "),
+                (Level::INFO, false) => self.writer.write_str("info: "),
+                (Level::DEBUG, false) => self.writer.write_str("debug: "),
+                (Level::TRACE, false) => self.writer.write_str("trace: "),
+            };
```
Comment (mystor) [inline inoculate/src/trace.rs:146]: I wish there was a nicer way to do conditional coloring without writing everything out twice. This is kinda unfortunate :'-(

File: inoculate/src/trace.rs [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+        }
+
+        if !self.is_empty {
```
Comment (mystor) [inline inoculate/src/trace.rs:149]: Could this be an `else`?

File: inoculate/src/trace.rs (hunk 1 above)
Comment (hawkw) [reply inoculate/src/trace.rs:27]: tracing logs to stdout by default; we should probably change it to stderr for consistency with cargo --- i'll do that now

File: inoculate/src/trace.rs (hunk 3 above)
Comment (hawkw) [reply inoculate/src/trace.rs]: 78cac5fc6ab5b8facd743a10905b2063a16b4ffd should make it a bit nicer --- i forgot that there _was_ a way to do that that i had written previously, lol :/

File: inoculate/src/trace.rs (hunk 2 above)
Comment (hawkw) [reply inoculate/src/trace.rs:135]: idk, I kind of like this. alternatively, we could just drop the special casing of "ing" and just always emit the cargo-style formatting for INFO events whose first word is less than the max length. which might actually be nicer, idk.

Comment (mystor) [reply inoculate/src/trace.rs]: I kinda like the idea of requiring a pattern like `Action: message goes here`, where a `:` after the first word is stripped and used to determine that it was intended to be a cargo style info event, but I don't feel a strong opinion about it.
